src/Polly.Contrib.Simmy.Specs/Behavior/InjectBehaviourAsyncWithOptionsSpecs.cs

```diff
@@ -109,6 +109,60 @@ public void Should_inject_behaviour_before_executing_user_delegate()
             injectedBehaviourExecuted.Should().BeTrue();
         }
 
+        #region BeforeInject
+        [Fact]
+        public async Task Should_call_before_inject_callback_before_injecting_behavior()
+        {
+            var beforeInjectExecuted = false;
+            var injectedBehaviourExecuted = false;
+
+            var policy = MonkeyPolicy.InjectBehaviourAsync(with =>
+                with.Behaviour(async () =>
+                {
+                    beforeInjectExecuted.Should().BeTrue();
+                    injectedBehaviourExecuted = true;
```

Comment:
make sure to either remove the async keyword from these delegates or `await` an `EmptyTask` so that the compiler doesn't complain about the lack of the `await` keyword:

```
var policy = MonkeyPolicy.InjectBehaviourAsync(with =>
                with.Behaviour(async () =>
                {
                    beforeInjectExecuted.Should().BeTrue();
                    injectedBehaviourExecuted = true;
                    await TaskHelper.EmptyTask;
                })
                .BeforeInject(async (context, cancellation) =>
                {
                    injectedBehaviourExecuted.Should().BeFalse();
                    beforeInjectExecuted = true;
                    await TaskHelper.EmptyTask;
                })
                .InjectionRate(0.6)
                .Enabled()
            );
```

**Note:** this applies to all the tests you added, you can see the warnings in the build: https://ci.appveyor.com/project/Polly-Contrib/simmy/builds/42270091

Reply:
Thank you. I did see the warnings, but haven't had time to address them yet.